v1/src/test/java/com/google/cloud/teleport/templates/TextToPubsubIT.java

```diff
@@ -81,22 +82,23 @@ public void testTextToTopic() throws IOException {
     // Act
     JobInfo info = launchTemplate(options);
     assertThat(info.state()).isIn(JobState.ACTIVE_STATES);
-    AtomicReference<PullResponse> records = new AtomicReference<>();
+    AtomicReference<List<String>> records = new AtomicReference<>(new ArrayList<>());
     Result result =
         new DataflowOperator(getDataflowClient())
-            .waitForConditionAndFinish(
+            .waitForCondition(
                 createConfig(info),
                 () -> {
                   PullResponse response =
                       pubsubResourceManager.pull(outputSubscription, expectedArtifacts.size());
-                  records.set(response);
-                  return response.getReceivedMessagesList().size() >= expectedArtifacts.size();
+                  if (response.getReceivedMessagesCount() > 0) {
+                    records.get().addAll(response.getReceivedMessagesList().stream()
+                        .map(receivedMessage -> receivedMessage.getMessage().getData().toStringUtf8())
+                        .collect(Collectors.toList()));
+                  }
+                  return records.get().size() >= expectedArtifacts.size();
                 });
     // TODO: Replace this check by truth subjects (like assertThatArtifacts(), assertThatRecords())
-    List<String> actualMessages =
-        records.get().getReceivedMessagesList().stream()
-            .map(receivedMessage -> receivedMessage.getMessage().getData().toStringUtf8())
-            .collect(Collectors.toList());
+    List<String> actualMessages = records.get();
     assertThat(result).isEqualTo(Result.CONDITION_MET);
     assertThat(actualMessages).containsAtLeastElementsIn(expectedArtifacts);
```

Comment:
Good catch. 

A bit of a nitpick, you don't need an AtomicReference here. Keeping the same reference is already atomic, so you can have `List<String> records` and just `records.addAll()` when doing a new pull

Reply:
Ok I honestly wasn't sure about that and didn't want to mess with the logic - just keep it from failing. I can edit that and make sure it still passes. Thanks.

Reply:
Yeah that's fine -- Java has some restrictions on what can be used from within the lambda not being reassigned.

So you can't have `PullResponse records` and keep reassigning it, that was the reason why we used an `AtomicReference`. But now that we can keep the same `List<String>` and will just aggregate new elements into the same list, we can just drop the `AtomicReference` as it may introduce confusion here for no benefits.